modules/storages/app/controllers/storages/admin/storages_controller.rb

```diff
@@ -179,7 +183,20 @@ def oauth_application(service_result)
   # update parameters are correctly set.
   def permitted_storage_params
     params
-      .require(:storages_storage)
+      .require(storage_provider_parameter_name)
       .permit('name', 'provider_type', 'host', 'oauth_client_id', 'oauth_client_secret')
   end
+
+  # TODO: Work out how to retrieve the storage provider resource name as it's based on the provider type
+  # PrimerForms implements Rails `form_with` which doesn't support overriding the form name as we would with
+  # `form_for`.
+  # See: https://github.com/opf/primer_view_components/blob/79fb58474771bd06946554f8325cd0b1bdd6dd31/app/helpers/primer/form_helper.rb#L7
+  #
+  def storage_provider_parameter_name
+    if OpenProject::FeatureDecisions.storage_primer_design_active?
+      :storages_nextcloud_storage
+    else
+      :storages_storage
+    end
+  end
```

Comment:
@opf/storages started discussions of introducing a FormObject to represent model CRUD operations

https://community.openproject.org/wp/50743